llvm/lib/Target/PowerPC/PPCInstrInfo.td

```diff
@@ -3345,7 +3345,7 @@ def LWZtocL : PPCEmitTimePseudo<(outs gprc:$rD), (ins tocentry32:$disp, gprc_nor
 def ADDIStocHA : PPCEmitTimePseudo<(outs gprc:$rD), (ins gprc_nor0:$reg, tocentry32:$disp),
                        "#ADDIStocHA",
                        [(set i32:$rD,
-                         (PPCtoc_entry i32:$reg, tglobaladdr:$disp))]>;
+                         (PPCtoc_entry tglobaladdr:$disp, i32:$reg))]>;
```

Comment:
Since we need to transform the PPCtoc_entry node into 2 new nodes for large code model we do the transformation directly in `PPCDAGToDAGISel::Select` and so the match pattern here isn't used and can be removed.

Reply:
Do we confirm the pattern is not used? @bzEq I saw there are LIT changes...

Reply:
Yes, the scheudling is changed. Since the machineinstr succeeds the attributes from the source SDAG node from the pattern. If the pattern is removed, the machineinstr loses these attributes. For `PPCtoc_entry`, the attributes are
```
SDNPMayLoad, SDNPMemOperand
```

Reply:
OK. I think marking ADDIStocHA without load/store attributes is the right behavior.